File: app/src/main/java/com/example/myapplication/data/WorkoutFeedbackRepository.kt
package com.example.myapplication.data

import com.example.myapplication.core.feedback.WorkoutDifficulty
import com.example.myapplication.core.feedback.WorkoutFeedback
import kotlinx.coroutines.flow.Flow

interface WorkoutFeedbackRepository {
    fun observeForGoal(goalId: Long): Flow<List<WorkoutFeedback>>

    suspend fun save(
        sessionId: Long,
        goalId: Long,
        completedEpochDay: Long,
        difficulty: WorkoutDifficulty,
    )
}

File: flutter/lib/data/repositories/workout_feedback_repository.dart
import '../../core/model/feedback_models.dart';

abstract class WorkoutFeedbackRepository {
  Stream<List<WorkoutFeedback>> observeForGoal(int goalId);

  Future<void> save({
    required int sessionId,
    required int goalId,
    required int completedEpochDay,
    required WorkoutDifficulty difficulty,
  });
}
